GameData/B9_Aerospace_HX/Patches/CRP/B9_Aerospace_HX-Engine_Instance.cfg

```diff
@@ -0,0 +1,357 @@
++PART[B9_Structure_HX_HPD1] // :NEEDS[CommunityResourcePack]
+{
+	@name = B9_Structure_HX_HPD2
+	@title = HX-HPD 2 Heavy Propulsion Device
+	@description = This variant of the HPD is a fission rocket and operates on Karborundum and Water. It remains quite powerful under intense atmosphere pressures where most engines may fail. The HPD 2 is a worthy candidate for engines that can bring a ship from the depthes of a gas giant... But who would be silly enough to measure these depths? Oh wait a minute...
+	
+	@EFFECTS
+	{
+		!running_lfo {}
+		!running_lava {}
+		!power_classic {}
+		!power_plasma {}
+		running_nswr
+		{
+			MODEL_MULTI_PARTICLE
+			{
+				name = flare
+				modelName = B9_Aerospace_HX/FX/explSpark
+				transformName = nozzle_transform
+				emission = 0.0 0.0
+				emission = 1.0 1.0
+				speed = 0.0 0.2
+				speed = 1.0 1.0
+			}
+			MODEL_MULTI_PARTICLE
+			{
+				name = stream
+				modelName = B9_Aerospace_HX/FX/explStream
+				transformName = nozzle_transform
+				emission = 0.0 0.0
+				emission = 1.0 1.0
+				speed = 0.0 0.2
+				speed = 1.0 1.0
+			}
+		}
+		power_nswr
+		{
+			AUDIO
+			{
+				name = heavy
+				channel = Ship
+				clip = sound_rocket_hard
+				volume = 0.05 0.0
+				volume = 1.0 3.0
+				pitch = 0.0 0.6
+				pitch = 1.0 0.9
+				loop = true
+			}
+			AUDIO
+			{
+				name = jet
+				channel = Ship
+				clip = sound_jet_low
+				volume = 0.0 0.0
+				volume = 0.5 0.5
+				pitch = 0.0 0.1
+				pitch = 0.5 0.3
+				loop = true
+			}
+		}
+	}
+	
+	!MODULE[MultiModeEngine] {}
+	@MODULE[ModuleEnginesFX],0
+	{
+		@engineID = Karborundum
+		@runningEffectName = running_nswr
+		@powerEffectName = power_nswr
+		@heatProduction = 500
+		@maxThrust = 24000
+		!PROPELLANT,* {}
+		PROPELLANT:NEEDS[CommunityResourcePack]
```

Comment:
Won't this end up with no propellants in CRP is not installed?

Reply:
I've had many issues where KSP hangs because an engine failed to compile because resource definitions for its propellants are missing. Here, KSP will not hang but the engine will be harmless to the startup process (although, by consequence, also useless, to reduce the incoming hard dependence on CRP. Perhaps I should make it hide from part selection in this case?)

Reply:
I think that would be a better option.  Could even be a `:NEEDS`

Reply:
Having the engine's even loading in depend on NEEDS, I decided against that as it could break ships in-flight if for whatever reason someone uninstalls CRP. The likelihood of this is actually quite low I think but it seemed a very odd choice.